proxy/m2_actions.c

```diff
@@ -1886,6 +1886,77 @@ enum http_rc_e action_content_delete (struct req_args_s *args) {
 	return _reply_m2_error (args, err);
 }
 
+static enum http_rc_e
+_m2_content_delete_many (struct req_args_s *args, struct json_object * jbody) {
+	guint max_request = OIO_CONTENT_DELETE_MANY_NB_MAX_REQ;
+	GError *err = NULL;
+	const char *url_path = oio_url_get(args->url, OIOURL_PATH);
+	json_object *jarray = NULL;
+	PACKER_VOID(_pack) { return m2v2_remote_pack_DEL (args->url); }
+
+	if (!oio_url_get(args->url, OIOURL_ACCOUNT)
+			|| !oio_url_get(args->url, OIOURL_NS))
+		return _reply_format_error(args,
+				BADREQ("Missing account or namespace"));
+
+	if (!json_object_object_get_ex(jbody, "contents", &jarray)
+			|| !json_object_is_type(jarray, json_type_array))
+		return _reply_format_error(args,
+				BADREQ("Invalid array of contents"));
+
+	GString *gresponse = g_string_sized_new(2048);
+	g_string_append(gresponse, "{\"contents\":[");
```

Comment:
Since you `return` abruptly after checking input array, you should allocate `gresponse` after, or you will get a memory leak.